framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -256,198 +272,86 @@ private void viewpointAdaptExecutable(
         // Then copy annotations from the viewpoint adapted type to methodType, if that annotation
         // is not on a type that was substituted for a type variable.
 
-        viewpointAdaptToContext(context, viewpointAdaptedType);
+        StringToJavaExpression stringToJavaExpr;
+        if (tree instanceof MethodInvocationTree) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atMethodInvocation(
+                                    stringExpr, (MethodInvocationTree) tree, factory.getChecker());
```

Comment:
Minor:  For some lambdas, you create a local variable with the casted value (maybe to avoid a cast on every execution of the lambda body?), and for some like this one you do not.